src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1770,6 +1771,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
             if (notification.HandledResponse) {
                 return;
             }
+            
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
```

Comment:
Note: even extracted from the ticket, the `client_id` property is not appropriate to determine whether an application was fully authenticated (don't forget that tokens issued by the authorization endpoint will have this property).

There's actually another property you can use: `OpenIdConnectConstants.Extra.ClientAuthenticated`. It will only be set if the token was issued by the token endpoint to a fully authenticated app (i.e if the developer calls `Validated()` but not `Skipped()`).